Content.Server/Chemistry/EntitySystems/ReagentDispenserSystem.cs

```diff
@@ -97,26 +97,38 @@ private void UpdateUiState(Entity<ReagentDispenserComponent> reagentDispenser)
                 var storageSlotId = ReagentDispenserComponent.BaseStorageSlotId + i;
                 var storedContainer = _itemSlotsSystem.GetItemOrNull(reagentDispenser.Owner, storageSlotId);
 
-                // Set label from manually-applied label, or metadata if unavailable
+                // Add volume remaining label and prepare solution contents string
+                FixedPoint2 quantity = 0f;
+                string solContents = "";
+                if (storedContainer != null && _solutionContainerSystem.TryGetDrainableSolution(storedContainer.Value, out _, out var sol))
+                {
+                    quantity = sol.Volume;
+                    solContents = string.Join("-", sol.Contents.Select(x =>
+                        {
+                            return (_prototypeManager.TryIndex(x.Reagent.Prototype, out ReagentPrototype? p) ? p.LocalizedName
+                            : Loc.GetString("reagent-dispenser-window-reagent-name-not-found-text"));
+                        })
+                    );
+                }
```

Comment:
Client should be the one sorting it.

Reply:
I moved the actual sorting client-side, but the code you highlighted is the (allegedly!) hacky label building. Do you think it would make more sense to move it to the client too?